src/fragments/lib/push-notifications/react-native/migrate_from_previous_version/migrate-from-previous-version.mdx

```diff
@@ -0,0 +1,170 @@
+
+```objective-c
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+  [AmplifyPushNotification didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
+}
+
+- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult result))completionHandler {
+  [AmplifyPushNotification didReceiveRemoteNotification:userInfo withCompletionHandler:completionHandler];
+}
+```
+
+</Block>
+<Block name="Android">
+
+### Update gradle scripts
+
+Amplify Push Notifications now requires a minimum target of Android SDK API level 24.
+
+1. Open `build.gradle` located inside the `/android` folder of your React Native project with a text editor.
+
+2. Update the `minSdkVersion` value in the `buildscript` block to at least `24`.
+
+```groovy
+buildscript {
+  ext {
+    ...
+    minSdkVersion = 24
+    ...
+  }
+}
+```
+
+3. Open the application `build.gradle` located inside the `/android/app` folder of your React Native project with a text editor.
+
+4. Remove firebase dependencies from the `dependencies` block.
+
+```diff
+dependencies {
+  ...
+-   implementation "com.google.firebase:firebase-core:15.0.2"
+-   implementation "com.google.firebase:firebase-messaging:15.0.2"
+  ...
+}
+```
+
+### Update the manifest
+
+5. Open `AndroidManifest.xml` located inside the `/android/app/src/main` folder.
+
+6. Remove the old service block from the `application`.
+
+```diff
+<application ... >
+  ...
+- <!--[START Push notification config -->
+- <service
+-   android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationMessagingService"
+-   android:exported="false">
+-   <intent-filter>
+-     <action android:name="com.google.firebase.MESSAGING_EVENT"/>
+-   </intent-filter>
+- </service>
+- <!-- [END Push notification config -->
+  ...
+</application>
+```
+
+</Block>
+</BlockSwitcher>
+
+## Update your application code
+
+### Add required polyfills
+
+You need to add the crypto.getRandomValues and URL polyfill to your application's entry point file (in most React Native apps this will be the top level `index.js`).
+
+```js
+// Example index.js
+import 'react-native-get-random-values';
+import 'react-native-url-polyfill/auto';
+...
+```
+
+### Remove Analytics (Optional)
+
+**If you do not** use Amplify Analytics in your application but have previously imported it in your code just to implement Amplify Push Notifications, you should be able to remove it now. Analytics is no longer required to be configured in your frontend application to use Amplify Push Notifications.
+
+### Replace APIs
+
+Please refer to the [Getting started](/lib/push-notifications/getting-started) portion of this guide to start learning more about the new Amplify Push Notifications.
+
+A brief, but imperfect, mapping of your APIs to their replacements is as follows:
```

Comment:
```suggestion
A brief mapping of your APIs to their replacements is as follows:
```

What's imperfect about it? Do customers need to know the nuance or does the paragraph that follows cover that concern?